ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/move_executor.rs

```diff
@@ -0,0 +1,305 @@
+            }
+            // Parse the Move.toml file to get the module name.
+            let move_toml_path = move_module_folder.join("Move.toml");
+            let move_toml_content = tokio::fs::read_to_string(&move_toml_path).await?;
+            let move_config: MoveConfig = toml::from_str(&move_toml_content).context(format!(
+                "Failed to parse Move.toml file at {:?}",
+                move_toml_path
+            ))?;
+            let package_name = move_config.package.name;
+
+            // Run the compiled Move Script.
+            let mut run_cmd = tokio::process::Command::new(APTOS_CLI_BINARY_NAME);
+            run_cmd
+                .arg("move")
+                .arg("run-script")
+                .arg("--compiled-script-path")
+                .arg(format!("build/{}/bytecode_scripts/main.mv", package_name))
+                .arg("--assume-yes");
+            match run_cmd.output().await {
+                Ok(output) => {
+                    if !output.status.success() {
+                        return Err(anyhow::anyhow!("Failed to run Move script: {:?}", output));
+                    }
+                    let aptos_cli_output: AptosCliOutput =
+                        match serde_json::from_slice(&output.stdout)
+                            .context("Failed to parse aptos output.")
+                        {
+                            Ok(output) => output,
+                            Err(e) => {
+                                // This means the
+                                return Err(anyhow::anyhow!(
+                                    "Failed to parse aptos output: {:?}",
+                                    e
+                                ));
+                            },
+                        };
+                    Ok(Some(aptos_cli_output.result.version))
+                },
+                Err(e) => Err(anyhow::anyhow!("Failed to run Move script: {:?}", e)),
+            }
+        }
+    }
+
+    // Internal methods.
+
+    /// Wait for the node to be ready.
+    async fn wait_if_not_ready(&self) -> anyhow::Result<()> {
+        let client = Client::new();
+        let start = std::time::Instant::now();
+        loop {
+            match client.get(&self.node_api_endpoint).send().await {
+                Ok(response) => {
+                    if response.status().is_success() {
+                        break;
+                    }
+                },
+                Err(_) => {
+                    if start.elapsed().as_secs() > NODE_READY_TIMEOUT_IN_SECS {
+                        return Err(anyhow::anyhow!(
+                            "Node is not ready after {} seconds.",
+                            NODE_READY_TIMEOUT_IN_SECS
+                        ));
+                    }
+                },
+            }
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+        }
+        Ok(())
+    }
+
+    /// Load the config from .aptos/config.toml
```

Comment:
This feels quite weird. There is a config for the CLI at `.aptos/config.yaml`, and now we're going to make another config just for this tool right alongside that called `.aptos/config.toml`? That feels really confusing. Using `.aptos` doesn't even necessarily seem valid, we only use that directory for files created by the CLI.

Reply:
I see in the code that the file is called `.aptos/config.yaml`. This is even worse if so, since we'd be clobbering the CLI config. This tool should not overwrite or alter the CLI config if it is not itself part of the CLI, we should use a separate config. I can see already that it uses the `default` profile for a separate purpose. This is not valid, the CLI by default already uses this profile for the CLI.